qa/os/src/test/java/org/elasticsearch/packaging/test/PackageTestCase.java

```diff
@@ -136,6 +136,22 @@ public void test32JavaHomeOverride() throws Exception {
         assertRunsWithJavaHome();
     }
 
+    public void test33RunsIfJavaNotOnPath() throws Exception {
+        assumeThat(distribution().hasJdk, is(true));
+
+        final Result readlink = sh.run("readlink /usr/bin/java");
+        assertThat(readlink.exitCode, equalTo(0));
```

Comment:
`Shell.run` already checks that `exitCode` is 0.

Reply:
Thanks, I pushed 4581efed952c9d6dd9712f5c12589da4812d925d.